include/nlohmann/detail/input/binary_reader.hpp

```diff
@@ -120,6 +124,139 @@ class binary_reader
     }
 
   private:
+
+    /*!
+    @return whether array creation completed
+    */
+    bool get_bson_cstr(string_t& result)
+    {
+        auto out = std::back_inserter(result);
+        while (true)
+        {
+            get();
+            if (JSON_UNLIKELY(not unexpect_eof()))
+            {
+                return false;
+            }
+            if (current == 0x00)
+            {
+                return true;
+            }
+            *out++ = static_cast<char>(current);
+        }
+
+        return true;
+    }
+
+    bool parse_bson_entries(bool is_array)
```

Comment:
Please document the parameter and return value.

Reply:
done